tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,180 @@
+
+        self.logger.info(
+            f"!SETTINGUP RP! extras:{self.redpanda._extra_rp_conf}")
+        super(ThroughputLimitsSnc, self).setUp()
+
+    def current_effective_node_quota(self) -> Tuple[int, int]:
+        metrics = self.redpanda.metrics_sample(
+            "quotas_snc_quota_effective",
+            metrics_endpoint=MetricsEndpoint.METRICS)
+
+        assert metrics, "Effecive quota metric is missing"
+        self.logger.debug(f"Samples: {metrics.samples}")
+
+        node_quota_in = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "ingress"
+            }).samples)
+        node_quota_eg = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "egress"
+            }).samples)
+        return node_quota_in, node_quota_eg
+
+    @cluster(num_nodes=3)
+    def test_configuration(self):
+        """
+        Test various configuration patterns, including extreme ones,
+        verify that it does not wreck havoc onto cluster
+        """
+
+        # TBD: parameterize to run under load or not
+
+        errors = 0
+        run_until = time.clock_gettime(time.CLOCK_MONOTONIC) + 15
+        # seconds
+        while time.clock_gettime(time.CLOCK_MONOTONIC) < run_until:
+
+            config_param = self.rnd.choice(list(self.ConfigProp))
+            config_value = self.get_config_parameter_random_value(config_param)
+            self.logger.debug(f"Setting {config_param.value} = {config_value}")
+            self.config[config_param] = config_value
+            self.rpk.cluster_config_set(config_param.value,
+                                        str(config_value or ''))
+
+            time.sleep(1)
+
+            def check_node_quota_metric(self, config_prop: self.ConfigProp,
+                                        effective_node_quota: int) -> int:
+                config_val = self.config[config_prop]
+                if config_val is None:
+                    return 0
+                expected_node_quota = config_val * len(self.redpanda.nodes)
+                if effective_node_quota == expected_node_quota:
+                    return 0
+                self.logger.error(
+                    f"Expected quota value mismatch. "
+                    f"Effective {effective_node_quota} != expected {expected_node_quota}. "
+                    f"Direction: {config_prop.name}")
+                return 1
+
+            effective_node_quota = self.current_effective_node_quota()
+            self.logger.debug(
+                f"current_effective_node_quota: {effective_node_quota}")
+            errors += check_node_quota_metric(
+                self, self.ConfigProp.QUOTA_NODE_MAX_IN,
+                effective_node_quota[0])
+            errors += check_node_quota_metric(
+                self, self.ConfigProp.QUOTA_NODE_MAX_EG,
+                effective_node_quota[1])
+
+        if errors != 0:
```

Comment:
`assert errors == 0, errors`

Reply:
or you could have `check_node_quota_metric` return strings on failure, keeping the error strings in a list and asserting the list is empty. So that when there is a failure you have all of the errors in the test output at the bottom instead of having to filter the log files for error logs